CenterForOpenScience/osf.io
# [QA Prep Required][Migration Required] Feature/Update AsPredicted form to the latest version [ENG-141]

## Purpose

The OSF offers the AsPredicted registration form as an option when creating a registration. 
AsPredicted has updated their registration form to be 9 questions, instead of 8 questions long. They have also modified the wording of some of the existing questions. We should remain up-to-date with registration forms.

## Changes
- Updated as-predicted schema to latest language with new questions
- Added new schema version to the db
- Modified how returning the latest version of a schema works - instead of just assuming version 2 is the latest version (this was hardcoded), return the latest of each name. (assuming schema names remain consistent between versions!)


## QA Notes

- Make some registrations ahead of time using AsPredicted schema.
- Verify afterwards that these registrations with old schema versions are still accessible (showing only 8 questions) and show up in search 
- Verify new registrations made with AsPredicted schema have 9 questions and likewise show up n search
- /v2/schemas/registrations/ shows latest AsPredictedSchema version 3. 

## Documentation



## Side Effects



## Ticket

https://openscience.atlassian.net/browse/ENG-141
